packages/icon/src/createGlyphComponent.tsx

```diff
@@ -3,6 +3,27 @@ import PropTypes from 'prop-types';
 import { SVGR, LGGlyph } from './types';
 import { css, cx } from '@leafygreen-ui/emotion';
 
+export const Size = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type Size = typeof Size[keyof typeof Size];
+
+export const sizeMap: Record<Size, number> = {
+  small: 14,
+  default: 16,
+  large: 20,
+  xlarge: 24,
+} as const;
+
+// We omit size here because we map string values for size to numbers in this component.
+export interface GlyphProps extends Omit<LGGlyph.ComponentProps, 'size'> {
```

Comment:
This might not be necessary now – I think this was from before Glyphs directly supported these string default sizes.

Reply:
I added this -- I think it is necessary so that we can overwrite the expected value of the size key.

Reply:
It doesn't need to be overwritten? `LGGlyph.ComponentProps` has the same type for size as this is overwriting. Additionally, the LGGlyph namespace should remain the single source of truth for these types.

Reply:
You're right -- sorry. This can be omitted.